src/libraries/System.Private.CoreLib/src/System/Span.cs

```diff
@@ -284,11 +284,18 @@ public void Fill(T value)
         {
             if (Unsafe.SizeOf<T>() == 1)
             {
-                // Special-case single-byte types like byte / sbyte / bool.
-                // The runtime eventually calls memset, which can efficiently support large buffers.
-                // We don't need to check IsReferenceOrContainsReferences because no references
-                // can ever be stored in types this small.
-                Unsafe.InitBlockUnaligned(ref Unsafe.As<T, byte>(ref _pointer.Value), Unsafe.As<T, byte>(ref value), (uint)_length);
+#if MONO
+                // Mono runtime's implementation of initblk performs a null check on the address.
+                // We'll perform a length check here to avoid passing a null address in the empty span case.
```

Comment:
Is that a compatibility thing? Should we change it?

Reply:
See comment at https://github.com/dotnet/runtime/pull/51365#issuecomment-821765770. I think I know where to make the change, but I was concerned that it would have broader implications and that I'd risk inadvertently breaking some other component. Putting the check here seemed the safest option. (This code had a check originally, so it's not a behavioral change from before this PR.)

Reply:
File an issue? Making the runtimes behave the same would be best if possible.

Reply:
Already filed - #51411 😄.

Reply:
Is that the same bug?